### ISSUE TITLE
Could you request a localStorage management api?

### ISSUE THREAD

**Issue by bughit** [ysard/cookie-quick-manager#38]
Would be nice to be able to view, protect, delete localStorage, as CookieKeeper was able to do.

Since they provide a cookie management api, hard to imagine they would refuse localStorage on principle , since they are conceptually similar.

**Comment by ysard** [comment]
Yes, I totally agree. This is all the more necessary as as explained in #37, these web storage solutions are used as invisible and unsecured cookies.

The fact that there is no API forces me to do the work of Mozilla via methods that I do not consider "clean":

- No possibility to query all the keys of the LocalStorage store (except for the sites currently loaded in the tabs),
- Mandatory injection of scripts in all the pages to iterate over keys and listen for changes.
- Important development of code that will no longer be useful if an API is integrated later in Firefox.

This point has already been discussed in the past and the proposal has not been retained:
https://bugzilla.mozilla.org/show_bug.cgi?id=1329745

The (debatable) arguments advanced were:
- Accessing it and manipulating it should be possible from content scripts.
- I do not think the complexity and overhead of this kind of API would be worthwhile for the little it adds beyond what is already available.
- The API cookie is not a fully accessible API, and is not fully accessible any other way. That's not the case for localStorage.

In the meantime, people have no easy way of knowing what sites visited are storing...

Currently, I think about the question but it does not have priority over the bug fixes in the interface and the current management of conventional cookies for the moment.
I will make a new proposal on Bugzilla; maybe the issue will be different now that Firefox Quantum is well developed.
